### ISSUE TITLE
Style/BracesAroundHashParameters when EnforcedStyle: context_dependent

### ISSUE THREAD

**Issue by EGiataganas** [bbatsov/rubocop#4205]
Here's a question. In my `.rubocop.yml`, I have enabled the following rule:

```ruby
Style/BracesAroundHashParameters:
  EnforcedStyle: context_dependent
```

Although when I am trying:

```ruby
Test.new(foo, { bar: 1 })
```

I am getting:

```ruby
Inspecting 1 file
C

Offenses:

C: Redundant curly braces around a hash parameter.
  Test.new(foo, { bar: 1 })
                ^^^^^^^^^^
```

## RuboCop version

```
$ rubocop -V
0.48.0 (using Parser 2.4.0.0, running on ruby 2.2.4 x86_64-darwin15)
```

Is this the expected behaviour with the `context_dependent` style? I think it's wrong complaining about the style of my example. Any ideas, feedback, please?

**Comment by bbatsov** [comment]
> Is this the expected behaviour with the context_dependent style? I think it's wrong complaining about the style of my example. Any ideas, feedback, please?

I think it won't complain only if you have hash literals for the final two params of a method. Guess the documentation of this cop is not good enough.

**Comment by bbatsov** [comment]
@EGiataganas I've extended the cop documentation a lot, so it should be pretty easy now to figure what each style is supposed to enforce.
